pkg/security/podsecuritypolicy/provider.go

```diff
@@ -449,3 +452,13 @@ func validateRuntimeClassName(actual *string, validNames []string) field.ErrorLi
 	}
 	return field.ErrorList{field.Invalid(field.NewPath("spec", "runtimeClassName"), *actual, "")}
 }
+
+func allowsVolumeType(allowedVolumes sets.String, fsType policy.FSType, volume api.Volume) bool {
+	if allowedVolumes.Has(string(fsType)) {
+		return true
+	}
+	if allowedVolumes.Has(string(policy.Secret)) && fsType == policy.Projected && utilfeature.DefaultFeatureGate.Enabled(features.BoundServiceAccountTokenVolume) && reflect.DeepEqual(*volume.VolumeSource.Projected, *serviceaccount.TokenVolumeSource) {
```

Comment:
this needs a comment to explain what is happening

Reply:
done